File: codewars/6kyu/count-characters-in-your-string/dduck-count-characters-in-your-string.py
# Count characters in your string
# https://www.codewars.com/kata/52efefcbcdf57161d4000091/

from unittest import TestCase


def count(string):
    result = {}
    for char in list(string):
        if char in result:
            result[char] = result[char] + 1
        else:
            result[char] = 1
    return result


TestCase().assertEqual(count('aba'), {'a': 2, 'b': 1})
TestCase().assertEqual(count(''), {})

File: codewars/6kyu/count-characters-in-your-string/dduck-count-characters-in-your-string.js
// count-characters-in-your-string
// https://www.codewars.com/kata/52efefcbcdf57161d4000091/

const count = (string) => {
  const map = {};
  string.split('').forEach((v) => map[v] = (map[v] || 0) + 1);
  return map;
};

console.log(count("aba"), { a: 2, b: 1 });
